py/server/deephaven/dtypes.py

```diff
@@ -202,6 +202,9 @@ def array(dtype: DType, seq: Sequence, remap: Callable[[Any], Any] = None) -> jp
     Raises:
         DHError
     """
+    if not isinstance(dtype, DType):
```

Comment:
Wait, we really need to do this? Python will let you just pass any type you want?

Reply:
yeah, that's Python, the idiomatic approach is to not check the type of the argument but maybe the value, but we have a  problem with the `raise` statement in the try/except block that intends to provide the Java class name in the message:
`
    except Exception as e:
        raise DHError(e, f"failed to create a Java {dtype.j_name} array.") from e
`